src/Dispatcher.cpp

```diff
@@ -195,6 +199,27 @@ void DispatcherBase::NotifyConnections(
   for (const auto& conn : m_connections) conn->NotifyIfActive(callback);
 }
 
+void DispatcherBase::SetConnector(Connector connector) {
+  std::vector<Connector> connectors;
+  connectors.push_back(connector);
+  SetConnector(std::move(connectors));
+}
+
+void DispatcherBase::SetConnector(std::vector<Connector>&& connectors) {
+  std::lock_guard<std::mutex> lock(m_user_mutex);
+  m_client_connectors = std::move(connectors);
+}
+
+void DispatcherBase::SetConnectorOverride(Connector connector) {
```

Comment:
This connector override thing seems strange to me. Why not just query all of the possible connectors for potential addresses to connect to?